test/unit/artefact/artefact_test.rb

```diff
@@ -0,0 +1,43 @@
+require_relative '../../test_helper'
+
+class Artefact::ArtefactTest < ActiveSupport::TestCase
+
+  context "taggable?" do
+
+    should "return true for artifacts not owned by publisher, smartanswers or testapp" do
+      Settings.expects(:untaggable_apps).returns(%w{ publisher smartanswers testapp }).at_least(1)
```

Comment:
It does, and in this case I didn't mention "preconditions" for this exact reason. This could be as simple as introducing a new `context` say:

```ruby
context 'with untaggable apps' do
  setup do
    Settings.stubs(untaggable_apps: %w(publisher smartanswers test app)
  end

  # all the shoulds that rely on those untaggable apps
end

context 'without untaggable apps' do
  setup do
    Settings.stubs(untaggable_apps: nil)
  end

  # the should(s) that doesn't 
end
```

Does that make sense? It cuts down on some of the duplication and just lends some natural grouping. But I'm trying not to be over-prescriptive here, just suggestions and certainly not merge-blockers :+1: